### PR TITLE
fix(case-law): bound decision dates at the ingestion boundary

### PR DESCRIPTION
**Pull request by jan-kubica**
`decisionDate` reached the `decision_date` column verbatim. Adapters normalize it differently or not at all, so a non-calendar day or an impossible year was stored as readily as a real date — the cz-regional seed fixture alone carries `2359-11-17`, `2417-11-17`, and `2944-04-30`.

`canonicalDecisionDate` (`apps/api/src/lib/dates.ts`) reduces a bare ISO date or an ISO datetime to its `YYYY-MM-DD` prefix and returns it only for a real calendar day whose year falls between 1800 and next year. `sanitizeResult` applies it, so every adapter, backfill, and replay path shares one rule at the declared ingestion boundary rather than each publisher boundary restating (or forgetting) it.

An unusable date drops to null rather than failing the row: a listed document is still worth storing. The pipeline emits `case_law.ingestion.decision_date_out_of_bounds` at warn with the truncated raw value, so the rejection is visible and the publisher's original shape stays recoverable from the row's metadata.

Tests: helper edge cases in `dates.test.ts` (bounds years, non-calendar days, datetime prefixes, garbage) and a `sanitizeResult` describe in `pipeline.test.ts` over production-shaped fixtures. pl-courts' own inline year check becomes redundant but harmless; the boundary check subsumes it.

### CODE REVIEW

File: apps/api/src/lib/legal-search/ingestion-normalization.ts [hunk 1]
```diff
@@ -57,6 +63,15 @@ export const sanitizeResult = (result: IngestionResult): IngestionResult => {
     );
   };
 
+  // A listed document must survive a bad date, so an unusable value is
+  // dropped to null instead of failing the row.
+  const boundDecisionDate = (raw: string | undefined): string | undefined => {
+    if (raw === undefined) {
+      return undefined;
+    }
+    return canonicalDecisionDate(raw) ?? undefined;
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply apps/api/src/lib/legal-search/ingestion-normalization.ts:72]: Agreed, fixed in 0c62b4e04b.

An update omits an undefined column, so the rejection had to stay distinguishable from an unstated date all the way to the write rather than collapsing into `undefined` at the boundary. `processDecisionAttempt` now derives a single `persistedDecisionDate` used by both write sites: a usable date is written, a stated-but-rejected one is `null` so the update clears the column, and an unstated one stays `undefined` so the row keeps what it has (an adapter that simply does not carry a date on a given observation must not erase a good one).

Covered by three cases in `processDecision — decision date on an existing row`, asserting the captured update payload for each of the three states.

File: apps/api/src/lib/dates.ts [hunk 2]
```diff
@@ -52,6 +52,62 @@ export const parseIsoDateLocal = (value: string): Date | null => {
...
+ * is not a real calendar day (e.g. "2024-02-30") or whose year falls outside
+ * `DECISION_YEAR_BOUNDS`. A date column takes a malformed year as readily as
+ * a correct one, so callers writing to one need this in front of the write.
+ */
+export const canonicalDecisionDate = (raw: string): string | null => {
+  const candidate = isoDatePrefix(raw);
+  if (candidate === null) {
+    return null;
+  }
+  const parsed = parseIsoDateLocal(candidate);
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply apps/api/src/lib/dates.ts]: Agreed, fixed in 0c62b4e04b. Confirmed the mechanism: under `TZ=Pacific/Apia`, `new Date(2011, 11, 30)` resolves to Dec 31, so the round-trip check in `parseIsoDateLocal` rejects a real Gregorian day.

`canonicalDecisionDate` no longer goes through `parseIsoDateLocal`. It round-trips the parts through `Date.UTC` and reads them back with `getUTC*`; UTC skips no days, so the same value is accepted everywhere. `parseIsoDateLocal` keeps its local-midnight contract unchanged, since that one exists precisely to land on the intended local calendar day for rendering.

To keep the two from drifting apart on the date shape itself, both now take their parts from one `isoDateParts` helper over the existing `ISO_DATE_PATTERN`; only the calendar they check against differs. Regression test pins `parseIsoDateLocal("2011-12-30") === null` next to `canonicalDecisionDate("2011-12-30") === "2011-12-30"` under that TZ, so the distinction cannot be flattened later by accident.

File: apps/api/src/lib/dates.ts [hunk 3]
```diff
@@ -52,6 +52,62 @@ export const parseIsoDateLocal = (value: string): Date | null => {
...
+ */
+const isoDatePrefix = (raw: string): string | null => {
+  if (raw.length === ISO_DATE_LENGTH) {
+    return raw;
+  }
+  const separator = raw.charAt(ISO_DATE_LENGTH);
+  if (separator !== "T" && separator !== " ") {
+    return null;
+  }
+  return raw.slice(0, ISO_DATE_LENGTH);
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply apps/api/src/lib/dates.ts]: Agreed, fixed in 0c62b4e04b. Taking the first ten characters on the strength of one separator character means `2024-03-05Tgarbage` is stored as a date the source never stated, which is exactly the kind of silent acceptance this guard exists to remove.

`isoDatePrefix` now matches the remainder against a full time-of-day pattern before extracting the prefix: `T` or a space, hour `00-23`, minute `00-59`, optional `:ss`, optional fraction, optional `Z` or `±hh:mm` offset. Anything else returns null, so the value drops with the same warning as any other unusable date. The pattern has no nested or overlapping quantifiers, so it stays linear on input length.

Tests pin both directions: `2024-03-05T`, `2024-03-05Tgarbage`, `2024-03-05T25:99:99Z` and `2024-03-05-extra` reject, while `T14:30`, `T14:30:59.123Z` and `T14:30:00+02:00` still canonicalize to `2024-03-05`.
